fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNavigationMiddlewareTest.kt

```diff
@@ -28,10 +28,12 @@ class ReviewQualityCheckNavigationMiddlewareTest {
     @Before
     fun setup() {
         browserStore = BrowserStore()
-        addTabUseCase = TabsUseCases.SelectOrAddUseCase(browserStore)
+        addTabUseCase = mockk(relaxed = true)
```

Comment:
Now that we've removed the direct context dependency from this middleware, do you think there's value in mocking this or should we use the real one and assert the tab size as before. Or do you think roboelectric wasn't the cause of the flakiness?

Reply:
I don't think robolectric was the cause behind the test flakiness, my assumption was the usecase didn't update the browser store in time for our assertion to be true.
Removing Robolectric is definitely a plus and I think mocking the usecase gets us a more concise and independent test.

Reply:
I see, in that case verification is fine 👍  `TabsUseCase` is tested anyway that it will open the tab.